api/api-artifacts-v1/src/main/java/com/thoughtworks/go/apiv1/artifacts/ArtifactsControllerV1.java

```diff
@@ -0,0 +1,119 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.artifacts;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.CrudController;
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.apiv1.artifacts.represernter.ArtifactConfigRepresenter;
+import com.thoughtworks.go.config.ArtifactConfig;
+import com.thoughtworks.go.config.exceptions.EntityType;
+import com.thoughtworks.go.server.service.EntityHashingService;
+import com.thoughtworks.go.server.service.ServerConfigService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+
+import static com.thoughtworks.go.api.util.HaltApiResponses.haltBecauseEtagDoesNotMatch;
+import static spark.Spark.*;
+
+@Component
+public class ArtifactsControllerV1 extends ApiController implements SparkSpringController, CrudController<ArtifactConfig> {
+
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private ServerConfigService serverConfigService;
+    private EntityHashingService entityHashingService;
+
+    @Autowired
+    public ArtifactsControllerV1(ApiAuthenticationHelper apiAuthenticationHelper, EntityHashingService entityHashingService, ServerConfigService serverConfigService) {
+        super(ApiVersion.v1);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.serverConfigService = serverConfigService;
+        this.entityHashingService = entityHashingService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.Artifacts.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+
+            get("", mimeType, this::show);
+            put("", mimeType, this::update);
+
+            // change the line below to enable appropriate security
+            before("", mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
+            // to be implemented
```

Comment:
Please move this to line 70 (above the get and put routes) and remove the comments